src/db-adapters/Mongoose/MongooseAdapter.js

```diff
@@ -202,16 +191,17 @@ export default class MongooseAdapter {
     const plural = this.inflector.plural;
 
     // Set up some data structures based on resourcesOrCollection
-    const resourceTypes = [];
-    const changeSets = {};
+    let resourceTypes = [];
+    let changeSets = {};
+
     const idOrIds = mapResources(resourceOrCollection, (it) => {
       changeSets[it.id] = it;
       resourceTypes.push(it.type);
       return it.id;
     });
 
-    const mode    = typeof idOrIds === "string" ? "findOne" : "find";
-    const idQuery = typeof idOrIds === "string" ? idOrIds : {"$in": idOrIds};
+    let idQuery = this.constructor.makeIdQuery(idOrIds);
+    let mode = typeof idQuery === 'string' ? 'findOne' : 'find';
```

Comment:
Why these changes? I.e. reordering mode and idQuery, switching to let, making the `typeof` on idQuery rather than `idOrIds`.

Given that `idQuery` and `mode` always go together, it probably makes sense to have have `makeIdQuery` return an array with a mode string and an id query, which could then be destructured. I.e.

```
const [mode, idQuery] = this.constructor.makeIdQuery(idOrIds)
```

(makeIdQuery—which I still really like as a utility method—might then be renamed to something like `getIdQueryType`)